File: proposed-projects/round-trip-experiment/reverse/20260909T153339Z_9de9fd66/20260910T090338Z_5eab4940/ported_implementation/src/rules-builder/parse-space.ts
export const parseSpace = (src: string, pos: number, newlineOk: boolean): number => {
  while (
    src[pos] === ' ' ||
    src[pos] === '\t' ||
    src[pos] === '#' ||
    (newlineOk && (src[pos] === '\r' || src[pos] === '\n'))
  ) {
    if (src[pos] === '#') {
      while (src[pos] !== undefined && src[pos] !== '\r' && src[pos] !== '\n') {
        pos += 1;
      }
    } else {
      pos += 1;
    }
  }
  return pos;
};

File: packages/gbnf-experiment/data/20260909T153339Z_9de9fd66/ported_implementation/gbnf/rules_builder/parse_space.py
def _at(src: str, pos: int) -> str:
    """Read a character, mirroring JS's out-of-bounds `undefined` as an empty string."""
    return src[pos] if 0 <= pos < len(src) else ""


def parse_space(src: str, pos: int, newline_ok: bool) -> int:
    while _at(src, pos) in (" ", "\t", "#") or (
        newline_ok and _at(src, pos) in ("\r", "\n")
    ):
        if _at(src, pos) == "#":
            while _at(src, pos) and _at(src, pos) not in ("\r", "\n"):
                pos += 1
        else:
            pos += 1
    return pos
